src/reducers/vis-state-updaters.js

```diff
@@ -1514,3 +1526,215 @@ export function updateAnimationDomain(state) {
     }
   };
 }
+
+/**
+ * Update the status of the editor
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {string} mode to set to editor to
+ * @return {Object} nextState
+ */
+export const setEditorModeUpdater = (state, {mode}) => ({
+  ...state,
+  editor: {
+    ...state.editor,
+    mode
+  }
+});
+
+/**
+ * Update editor features
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {[Object]} features to store
+ * @return {Object} nextState
+ */
+export function setFeaturesUpdater(state, {features = []}) {
+
+  const lastFeature = features.length && features[features.length - 1];
+
+  let newState = {
+    ...state,
+    editor: {
+      ...state.editor,
+      features,
+      mode: lastFeature && lastFeature.properties.isClosed ?
+        EDITOR_MODES.EDIT : state.editor.mode
+    }
+  };
+
+  // Retrieve existing feature
+  const {selectedFeature} = state.editor;
+
+  // If no feature is selected we can simply return since no operations
+  if (!selectedFeature) {
+    return newState;
+  }
+
+  // TODO: check if the feature has changed
+  const feature = features.find(f => f.id === selectedFeature.id);
+
+  if (!feature) {
+    return newState;
+  }
+
+  const filterIdx = getFilterIdxByFeatureId(newState, selectedFeature.id);
+
+  // no filters found
+  if (filterIdx === -1) {
+    return newState;
+  }
+
+  const filter = state.filters[filterIdx];
+
+  const newFilter = updatePolygonFilter(filter, feature);
+
+  const newFilters = Object.assign([...state.filters], {[filterIdx]: newFilter});
+
+  newState = {
+    ...state,
+    datasets: applyFiltersToDatasets(newFilter.dataId, newState.datasets, newFilters, newState.layers),
+    filters: newFilters
+  };
+
+  return updateAllLayerDomainData(newState, Object.keys(newState.datasets));
+}
+
+/**
+ * Set the current selected feature
+ * @memberof uiStateUpdaters
+ * @param {Object} state `uiState`
+ * @param {[Object]} features to store
+ * @return {Object} nextState
+ */
+export const setSelectedFeatureUpdater = (state, {selectedFeatureId}) => ({
```

Comment:
instead of save the id in reducer, why not save the entire feature, that way it easier to literate over its property, also nicer to pass the entire feature around when deleting it
https://github.com/keplergl/kepler.gl/commit/7f927840ab976305eee3200d3aedb903ab60c0df#diff-3c42abf857bb61c5f6578c72771405d2R1600